src/pages/SignIn.jsx

```diff
@@ -1,20 +1,87 @@
-import React from 'react';
-import { Link } from 'react-router-dom';
+import React, {useContext, useEffect, useRef, useState} from 'react';
+import {Link} from 'react-router-dom';
+import {AuthContext} from "../context/AuthContext";
+import axios from "axios";
 
 function SignIn() {
-  return (
-    <>
-      <h1>Inloggen</h1>
-      <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ab alias cum debitis dolor dolore fuga id molestias qui quo unde?</p>
-
-      <form>
-        <p>*invoervelden*</p>
-        <button>Inloggen</button>
-      </form>
-
-      <p>Heb je nog geen account? <Link to="/signup">Registreer</Link> je dan eerst.</p>
-    </>
-  );
+    const contextContent = useContext(AuthContext);
+    const abortController = new AbortController();
+    const isMounted = useRef(false);
+    const endpoint = "http://localhost:3000/login"
+
+    const [formState, setFormState] = useState({
+        email: "",
+        password: "",
+        signal: abortController.signal,
+    })
+
+    const [emailValue, setEmailValue] = useState("")
+    const [isSubmitted, toggleIsSubmitted] = useState(false);
+
+    function handleFormChange(e) {
+        setFormState({
+            ...formState,
+            [e.target.name]: e.target.value,
+        })
+    }
+
+    useEffect(() => {
+
+        if (isMounted.current) {
+            async function submit() {
+                try {
+                    const response = await axios.post(endpoint, formState);
+                    contextContent.logInHandler(response.data.accessToken);
+                } catch (e) {
+                    console.error(e);
+                }
+
+            }
+
+            void submit();
+
+            return function cleanup() {
+                abortController.abort();
+            }
+        }
+
+        isMounted.current = true;
+
+    }, [isSubmitted]);
+
+    function handleSubmit(e) {
+        e.preventDefault();
+        toggleIsSubmitted(true);
+    }
```

Comment:
Ik zou alleen de cleanup functie teruggeven in de useEffect, omdat je precies weet wanneer je de submit wil doen, verwacht ik deze ook in de handleSubmit functie (aangeroepen te worden)
```suggestion
    async function submit() {
        try {
            const response = await axios.post(endpoint, formState);
            contextContent.logInHandler(response.data.accessToken);
        } catch (e) {
            console.error(e);
        }
    }
    
    useEffect(() => {
        return function cleanup() {
            abortController.abort();
        }
    }, [isSubmitted]);

    function handleSubmit(e) {
        e.preventDefault();
        submit();
    }
```